### PR TITLE
add multiple device documentation

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

**Comment by itkg-bfouche** [comment]
:+1:

File: en/developer_guide/multiple_device.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+* The web device is the root one
+* If there is no android template, it will look for the phone one
+* If there is no phone template, it will look for the main one
+* If there is no tablet template, it will look for the main one
+
+Template engine
+---------------
+
+In order to swith automatically between all the template, we had to add some functionality in the template engine.
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: switch
templates
to the template engine

File: en/developer_guide/multiple_device.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+If you want to use another template engine, you need to implement these logic. We provide you with all the modification
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: these logics / this logic
modifications

File: en/developer_guide/multiple_device.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+we made in twig as an exemple.
+
+Twig exemple
+------------
+
+First we have extended both TwigEngine and TimedTwigEngine in the front bundle :
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: FrontBundle

File: en/developer_guide/multiple_device.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    FrontBundle\Twig\OrchestraTimedTwigEngine
+    FrontBundle\Twig\OrchestraTwigEngine
+
+Use the ``FrontBundle\Twig\Renderable`` trait which override the ``render()`` method by adding the device name
+in the template name
+
+.. code-block:: php
+
+    if (strstr($name, 'twig'))
+    {
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: psr2

File: en/developer_guide/multiple_device.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+------------
+
+First we have extended both TwigEngine and TimedTwigEngine in the front bundle :
+
+.. code-block:: php
+
+    FrontBundle\Twig\OrchestraTimedTwigEngine
+    FrontBundle\Twig\OrchestraTwigEngine
+
+Use the ``FrontBundle\Twig\Renderable`` trait which override the ``render()`` method by adding the device name
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: overrides

File: en/developer_guide/multiple_device.rst [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+in the template name
+
+.. code-block:: php
+
+    if (strstr($name, 'twig'))
+    {
+        return str_replace('html.twig', $device . '.html.twig', $name);
+    }
+
+Then, you declare the service based on the new class and put the templating alias :
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: "put the templating alias" -> i don't get what you mean

File: en/developer_guide/multiple_device.rst [hunk 7]
```diff
@@ -0,0 +1,89 @@
+Multiple device
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: devices ? multi-devices ?

File: en/developer_guide/multiple_device.rst [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+===============
+
+Prerequisite
+------------
+
+To enable the multi-device rendering, we assume that the server in front of Php (Varnish, Apache2, Nginx, ...)
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: "server in front of php" -> Are we speaking of the server running php, or a reverse proxy ? Or do you mean PHP-FPM is running behind a webserver ? The configuration is unclear to me.

File: en/developer_guide/multiple_device.rst [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        open_orchestra_front.twig.orchestra_twig_engine:
+            class: %open_orchestra_front.twig.orchestra_twig_engine.class%
+            arguments:
+                - @twig
+                - @templating.name_parser
+                - @templating.locator
+                - @request_stack
+                - %open_orchestra_front.devices%
+
+You can have a look on `this pull request`_ handling this modification
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: I don't think a PR should be referred to in the docs, as it basically says "go read the code", and can be out of date lateR.

File: en/developer_guide/multiple_device.rst [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+Prerequisite
+------------
+
+To enable the multi-device rendering, we assume that the server in front of Php (Varnish, Apache2, Nginx, ...)
+ implements the Wurfl librairy and is able to add a request header with the device type.
+
+The device name will be used in the configuration and in the template name following the rule :
+
+* main template : template.html.twig
+* phone template : template.phone.html.twig
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: The naming is unclear here. Is "phone" the name of a device ? Is it parametrable ? Even after further reading I'm not sure i get it.

File: en/developer_guide/multiple_device.rst [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    #app\config\config.yml
+    open_orchestra_front:
+        devices:
+            web: ~
+            tablet:
+                parent: web
+            phone:
+                parent: web
+            android:
+                parent: phone
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: Are these devices names hard coded ? If so, maybe make it clear before. Can I add my own device name ? How ?

File: en/developer_guide/multiple_device.rst [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+Template engine
+---------------
+
+In order to swith automatically between all the template, we had to add some functionality in the template engine.
+
+If you want to use another template engine, you need to implement these logic. We provide you with all the modification
+we made in twig as an exemple.
+
+Twig exemple
```
Comment (juchi) [inline en/developer_guide/multiple_device.rst]: example

**Comment by juchi** [comment]
General note : I think we could use a short intro explaining how the detection of the device and the rendering works.

**Comment by juchi** [comment]
:+1:

**Pull request merged**